mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerPerformer.java

```diff
@@ -109,31 +109,24 @@ public ReIndexerPerformer(MailboxManager mailboxManager,
         return handleMessageReIndexing(mailboxSession, mailbox, uid);
     }
 
-    private Task.Result reIndex(List<MailboxPath> mailboxPaths, MailboxSession mailboxSession, ReprocessingContext reprocessingContext) throws MailboxException {
-        return wrapInGlobalRegistration(mailboxSession,
-            globalRegistration -> handleMultiMailboxesReindexingIterations(mailboxPaths, globalRegistration, reprocessingContext));
-    }
-
-    private Task.Result handleMultiMailboxesReindexingIterations(List<MailboxPath> mailboxPaths, GlobalRegistration globalRegistration,
-                                                                 ReprocessingContext reprocessingContext) {
-        return mailboxPaths.stream()
-            .map(globalRegistration::getPathToIndex)
-            .flatMap(OptionalUtils::toStream)
-            .map(path -> {
+    private Task.Result reIndex(Stream<MailboxId> mailboxIds, ReprocessingContext reprocessingContext) {
+        return mailboxIds
+            .map(mailboxId -> {
                 try {
-                    return reIndex(path, reprocessingContext);
+                    return reIndexSingleMailbox(mailboxId, reprocessingContext);
                 } catch (Throwable e) {
-                    LOGGER.error("Error while proceeding to full reindexing on {}", path.asString(), e);
+                    LOGGER.error("Error while proceeding to full reindexing on {}", mailboxId, e);
                     return Task.Result.PARTIAL;
                 }
             })
             .reduce(Task::combine)
             .orElse(Task.Result.COMPLETED);
     }
 
-    private Task.Result reIndex(MailboxPath path, MailboxSession mailboxSession, ReprocessingContext reprocessingContext) throws MailboxException {
-        LOGGER.info("Intend to reindex {}", path.asString());
-        Mailbox mailbox = mailboxSessionMapperFactory.getMailboxMapper(mailboxSession).findMailboxByPath(path);
+    private Task.Result reIndexSingleMailbox(MailboxId mailboxId, ReprocessingContext reprocessingContext) throws MailboxException {
+        LOGGER.info("Intend to reindex {}", mailboxId);
```

Comment:
s/mailboxId/mailboxId.serialize() ?

Reply:
Okay but I don't want just to have just a UUID that could be anything, it should be clear it is a mailbox